DOCS-549: Updated info about keywords case-sensivity. (#9626)
ClickHouse/ClickHouse: docs/en/query_language/syntax.md

## Keywords {#syntax-keywords}

> Keywords are not case-sensitive when they correspond to:

Reviewer: ```suggestion
Keywords are case-insensitive when they correspond to:
```

> - SQL standard. For example, `SELECT = select`, `AS = as`.

Reviewer: ```suggestion
- SQL standard. For example, `SELECT`,  `select` and `SeLeCt` are all valid.
```

> - Implementation in some popular DBMS (MySQL or Postgres). For example, `DateTime = datetime`.

Reviewer: ```suggestion
- Implementation in some popular DBMS (MySQL or Postgres). For example, `DateTime` is same as  `datetime`.
```

> Case-sensitivity of data type names you can check in the `system.data_type_families` table.

Reviewer: ```suggestion
Whether data type name is case-sensitive can be checked in the `system.data_type_families` table.
```

link to system table?

Author: It is not described. I'll do it later.

> All other keywords (column names, functions, and so on), in contrast to standard SQL, are case-sensitive.

Reviewer: ```suggestion
In contrast to standard SQL all other keywords (including functions names)  are **case-sensitive**.
```
column name is an identifier, not a keyword
